File: 977. Squares of a Sorted Array/SquaresOfASortedArray.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        
        l, r, w = 0, len(nums)-1, len(nums)-1
        output = [0]*len(nums)
        
        while l <= r:
            
            if abs(nums[l]) > abs(nums[r]):
                output[w] = pow(nums[l],2)
                l += 1;
            else:
                output[w] = pow(nums[r],2)
                r -= 1
            
            w -= 1
        
        return output

File: 977. Squares of a Sorted Array/SquaresOfASortedArray.cs
public class Solution {
    public int[] SortedSquares(int[] nums) {
        
        int[] output = new int[nums.Length];
        
        int l = 0, r = nums.Length-1, w = r; 
        
        while (l <= r) {
            
            if (Math.Abs(nums[l]) > Math.Abs(nums[r])) {
                output[w--] = nums[l]*nums[l];
                l++;
            }
            else {
                output[w--] = nums[r]*nums[r];
                r--;
            }
              
        }
        return output;
    }
}
